src/lib.rs

```diff
@@ -507,10 +509,10 @@ macro_rules! impl_slice_range {
 #[macro_export]
 macro_rules! impl_slice_ranges {
     ( $fragment_type:ty ) => {
-        impl_slice_range! {$fragment_type, Range<usize>}
-        impl_slice_range! {$fragment_type, RangeTo<usize>}
-        impl_slice_range! {$fragment_type, RangeFrom<usize>}
-        impl_slice_range! {$fragment_type, RangeFull}
+        impl_slice_range! {$fragment_type, Range<usize>, |_| false }
+        impl_slice_range! {$fragment_type, RangeTo<usize>, |_| false }
```

Comment:
Isn't there room for optimization here too? I would say, if `range.end == fragment.len()`? (real question as I am not sure)

Reply:
Sure, but then I would have to pass `fragment` to the lambda, and it would make the code slightly harder to read. I don't think nom even uses `RangeTo` or `Range`, anyway.

Reply:
OK, seems good. We can change it anyway later.